Adds a "Hide Member Items" checkbox to the ABC Alchemy Panel

Alexalted opened the pull request:
Adding a "Hide Member Items" checkbox



Adds a "Hide Member Items" checkbox to the Panel, when refreshed/searched again the list updates to hide items that are tags as "isMember".
(Info taken from https://static.runelite.net/runelite-api/apidocs/net/runelite/api/ItemComposition.html#isMembers() )

Preview images:
![image](https://github.com/user-attachments/assets/55324bd6-e713-4693-a156-85e4750f7dd6)
![image](https://github.com/user-attachments/assets/bc4cdef9-865a-4d7f-ad77-51ad1745001a)

Alexalted committed:
Added "Filter Member Items" option, button is weird though.

Alexalted committed:
Added "Filter Member Items" option, button is weird though.

Alexalted committed:
Merge remote-tracking branch 'origin/master'

Alexalted committed:
Removed unnecessary import

Alexalted committed:
Changed "JToggleButton" to "JCheckBox", looks better

vartan commented:
This looks pretty good. I just have one nitpick: shouldn't it say "Members' item"? and similarly the instance variable AlchItem.isMember should be isMembers. could you update those things and I'll merge it in. thanks!

Alexalted commented:
Sure, I'll change it. 👍

I think for the menu I thought the equivalent on the other side would be for example "Hide _Free to Play_ Item**s**" instead of "Hide _Free to Play_**'s** item"? 🤔

But the instance variable makes sense to me 👌

Alexalted committed:
changed label display text & renamed usage of variable from "isMember" to "isMembers"

Alexalted commented:
@vartan The label gets slightly cut off when changed to "Hide Members' Item:";
Is that fine? (that's the commit I just made above)

<img width="283" height="424" alt="image" src="https://github.com/user-attachments/assets/ad12bee9-995e-4db7-bc75-c32c8a20722f" />

As a band-aid solution, I could remove the ":" at the end?
<img width="281" height="187" alt="image" src="https://github.com/user-attachments/assets/3e3e644d-63fc-49e3-a6d8-d376a280c3a9" />

Or change the label entirely to something else, i.e.
```Hide P2P Items:```
```Show F2P Items:```
```Show F2P Only:```

It's just this line of code here:
https://github.com/vartan/abc-alch/blob/ff4ada9ca646556648275b239444dae3a0a9bf8f/src/main/java/com/vartan/abc/AbcAlchPanel.java#L106

I'm not super familiar with JPanel, but there might be a way for someone else to adjust the "columns" of the "GridLayout" to allow for text overrun and/or push the checkbox to be aligned to the right.

Alexalted committed:
renamed checkbox "filterMemberItems" to "includeMemberItems";
Label update to "Members' Items" and 'checked' by default. When un-checked, it'll hide Members' items.

Alexalted committed:
changed JCheckBox creation to be "selected: true" by default
